http.go

```diff
@@ -207,41 +263,51 @@ func NewReverseProxy(config *Config, db *sql.DB) *httputil.ReverseProxy {
 
 		var resBody []byte
 		if res.Body != nil {
-			resBody, _ = io.ReadAll(res.Body)
-			res.Body = io.NopCloser(bytes.NewBuffer(resBody))
-			log.Printf("Response body length: %d", len(resBody))
+			var err error
+			resBody, err = io.ReadAll(res.Body)
+			if err != nil {
+				log.Printf("Error reading response body: %v", err)
+				return err
+			}
+			res.Body.Close() // Close the original body
+
+			// Check if this is a response where we need to modify fee percentages
+			apiKey, _ := res.Request.Context().Value("api_key").(string)
+			extraFee := getExtraFee(postgres, apiKey)
+			if extraFee != nil && res.Request.Method == "GET" {
+				switch res.Request.URL.Path {
+				case "/v2/swap/submarine", "/v2/swap/reverse", "/v2/swap/chain":
```

Comment:
Same here, there's no real point getting the fees if it's not one of these GET paths